src/common/LoadingBanner.js

```diff
@@ -0,0 +1,92 @@
+/* @flow strict-local */
+
+import React, { PureComponent } from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import type { Dispatch } from '../types';
+import { connect } from '../react-redux';
+import { getLoading } from '../selectors';
+import { Label, LoadingIndicator } from '.';
+import type { ThemeColors } from '../styles';
+import { ThemeContext } from '../styles';
+
+const key = 'LoadingBanner';
+
+const styles = StyleSheet.create({
+  block: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: 'hsl(6, 98%, 57%)',
+  },
+  none: { display: 'none' },
+});
+
+type SelectorProps = $ReadOnly<{|
+  loading: boolean,
+|}>;
+
+type Props = $ReadOnly<{|
+  spinnerColor?: 'black' | 'white' | 'default',
+  textColor?: string,
+  backgroundColor?: string,
+
+  dispatch: Dispatch,
+  ...SelectorProps,
+|}>;
+
+/**
+ * Displays a notice that the app is connecting to the server.
+ * Rendered when session.loading is true (as of 2020/02, only
+ * during the /register request).
+ *
```

Comment:
I appreciate the date marker for the uncertainty of how this may have changed in the future. :-)

The strategy I generally prefer, though, is to go perhaps a bit further:
* Say less here: perhaps like "Display a notice that the app is connecting to the server, when appropriate."
  * The idea is to only say what's in the interface of this code -- what it promises, for other code to rely on -- and not the specifics of the implementation choices it makes in order to live up to that interface.
* Then for the implementation choices, make them as clear as possible *inside* the code.
  * If the reader wants to know the details of how this code tries to live up to the "when appropriate" promise, they'll want to look at the code in any case, and the code should make their questions as easy to answer as it can.
  * Sometimes this involves comments.

Reply:
> Then for the implementation choices, make them as clear as possible inside the code.

I think we landed on LoadingState being the appropriate place for this comment; I'll add one there.